dts/bindings/sensor/memsic,mc3419.yaml

```diff
@@ -21,3 +21,47 @@ properties:
       has two interrupt pins.By default the interrupt are routed
       to interrupt pin 1, by enabled this property interrupt are
       routed to interrupt pin 2.
+
+  lpf-fc-sel:
+    type: int
+    default: 0
+    enum:
+      - 0
+      - 9
+      - 10
+      - 11
+      - 13
+    description: |
+      Select LPF cut-off frequency.
+      Possible values and corresponding values are listed below
+      0 => disable LPF
+      9 => LPF with fc set to IDR/4.255
+      10 => LPF with fc set to IDR/6
+      11 => LPF with fc set to IDR/12
+      13 => LPF with fc set to IDR/16
+      Default is LPF disabled (reset value of the register).
+
+  decimation-rate:
+    type: int
+    default: 0
+    description: |
+      This helps in producing slower Output Data Rate (ODR) from given
+      Input Data Rate (IDR). Possible values and corresponding decimation
+      rate are listed below
+      0  => ODR = IDR
+      1  => ODR = IDR / 2
+      2  => ODR = IDR / 4
+      3  => ODR = IDR / 5
+      4  => ODR = IDR / 8
+      5  => ODR = IDR / 10
+      6  => ODR = IDR / 16
+      7  => ODR = IDR / 20
+      8  => ODR = IDR / 40
+      9  => ODR = IDR / 67
+      10 => ODR = IDR / 80
+      11 => ODR = IDR / 100
+      12 => ODR = IDR / 200
+      13 => ODR = IDR / 250
+      14 => ODR = IDR / 500
+      15 => ODR = IDR / 1000
```

Comment:
I didn't get it. can you provide more details.
I'm new to open source contribution so forgive me if it is anything obvious.

Reply:
you're forcing the users to introduce register values, not actual rate.

Reply:
@gmarull that is because, i am unable to find pattern to easily convert real_decimation_rate to decimation_rate_cfg_value using a formula.
And if we still want to achieve that, we'll have to use a const array and trace it manually. that will have a unnecessary 16 length array to have ultimately use only single one, while rest 15 will be redundant. both code size and complexity will increase.

Let me know is there is a simple way, e.g. if it is possible to define enum variable, and use it's index in the code using some macro. Or something similar.
Otherwise i would argue a developer will relate it using dts documentation or datasheet.

Reply:
@gmarull nevermind, i have found couple of references in dts/bindings/sensor with enum. and refactored it using a header file. please review